File: MetaBrain/confidence/safe-display.ts
import type { ClinicalConfidenceFlags, ConfidencePolicy, HallucinationRisk, SafeDisplayResult } from "./types";

export function evaluateSafeDisplay(input: {
  confidenceScore: number;
  uncertaintyScore: number;
  hallucinationRisk: HallucinationRisk;
  escalationRecommended: boolean;
  flags: ClinicalConfidenceFlags;
  policy: ConfidencePolicy;
}): SafeDisplayResult {
  const restrictions: string[] = [];
  if (input.confidenceScore < input.policy.minimumSafeConfidence) restrictions.push("safe_summary_only");
  if (input.uncertaintyScore >= input.policy.highUncertaintyThreshold) restrictions.push("explicit_uncertainty_required");
  if (input.hallucinationRisk.escalation_required) restrictions.push("human_review_recommended");

  const unsafe = input.flags.safeDisplayEnabled && input.flags.blockingEnabled && input.escalationRecommended;

  return {
    safe_to_display: !unsafe,
    unsafe_to_display: unsafe,
    display_restrictions: restrictions,
  };
}

File: MB-Secretaria/confidence_py/safe_display.py
from .types import ClinicalConfidenceFlags, ConfidencePolicy, HallucinationRisk, SafeDisplayResult


def evaluate_safe_display(
    confidence_score: float,
    uncertainty_score: float,
    hallucination_risk: HallucinationRisk,
    escalation_recommended: bool,
    flags: ClinicalConfidenceFlags,
    policy: ConfidencePolicy,
) -> SafeDisplayResult:
    restrictions: list[str] = []
    if confidence_score < policy.minimum_safe_confidence:
        restrictions.append("safe_summary_only")
    if uncertainty_score >= policy.high_uncertainty_threshold:
        restrictions.append("explicit_uncertainty_required")
    if hallucination_risk.escalation_required:
        restrictions.append("human_review_recommended")
    unsafe = flags.safe_display_enabled and flags.blocking_enabled and escalation_recommended
    return SafeDisplayResult(safe_to_display=not unsafe, unsafe_to_display=unsafe, display_restrictions=restrictions)
